RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1481 @@
+        float value = 0.0f;
+        if (sample_over10ts == its_start) {
+          value = bin_0_start == -1
+                      ? accVarLenIdxMinusOneVec[distTo25ns_start] + factor * diffVarItvlIdxMinusOneVec[distTo25ns_start]
+                      : accVarLenIdxZeroVec[distTo25ns_start] + factor * diffVarItvlIdxZeroVec[distTo25ns_start];
+        } else if (sample_over10ts > its_start) {
+          int const bin_idx = distTo25ns_start + 1 + (sample_over10ts - its_start - 1) * ns_per_bx + bin_0_start;
+          value = acc25nsVec[bin_idx] + factor * diff25nsItvlVec[bin_idx];
+        }
+        return value;
+      }
+
+      // TODO: provide constants from configuration
+      // from RecoLocalCalo/HcalRecProducers/python/HBHEMahiParameters_cfi.py
+      constexpr int nMaxItersMin = 50;
+      constexpr int nMaxItersNNLS = 500;
+      constexpr double nnlsThresh = 1e-11;
+      constexpr float deltaChi2Threashold = 1e-3;
+
+      // from RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc
+      ALPAKA_FN_ACC float get_raw_charge(double const charge,
+                                         double const pedestal,
+                                         float const* shrChargeMinusPedestal,
+                                         float const parLin1,
+                                         float const parLin2,
+                                         float const parLin3,
+                                         int32_t const nsamplesForCompute,
+                                         int32_t const soi,
+                                         int const sipmQTSShift,
+                                         int const sipmQNTStoSum,
+                                         float const fcByPE,
+                                         int32_t const lch,
+                                         bool const isqie11) {
+        float rawCharge;
+
+        if (!isqie11)
+          rawCharge = charge;
+        else {
+          int const first = std::max(soi + sipmQTSShift, 0);
+          int const last = std::min(soi + sipmQNTStoSum, nsamplesForCompute);
+          float sipmq = 0.0f;
+          for (auto ts = first; ts < last; ts++)
+            sipmq += shrChargeMinusPedestal[lch * nsamplesForCompute + ts];
+          auto const effectivePixelsFired = sipmq / fcByPE;
+          auto const factor = compute_reco_correction_factor(parLin1, parLin2, parLin3, effectivePixelsFired);
+          rawCharge = (charge - pedestal) * factor + pedestal;
+
+#ifdef HCAL_MAHI_GPUDEBUG
+          printf("first = %d last = %d sipmQ = %f factor = %f rawCharge = %f\n", first, last, sipmq, factor, rawCharge);
+#endif
+        }
+        return rawCharge;
+      }
+
+      // Tried using lambda, but strange error from  with nvcc
+      inline constexpr bool TSenergyCompare(std::pair<unsigned int, float> a, std::pair<unsigned int, float> b) {
+        return a.second > b.second;
+      };
+
+      class kernel_prep1d_sameNumberOfSamples {
```

Comment:
Class names should start with upper case letter (also elsewhere)

Reply:
Done https://github.com/cms-sw/cmssw/commit/98367cbd606553c45f30a100a438bedbe880ae55